lib/options-manager.js

```diff
@@ -257,112 +272,146 @@ Transform an XO options into ESLint compatible options:
 		if (options.cwd && resolveFrom.silent(options.cwd, 'eslint-plugin-react')) {
-			config.plugins = config.plugins.concat('react');
-			config.rules['react/jsx-indent-props'] = ['error', spaces];
-			config.rules['react/jsx-indent'] = ['error', spaces];
+			config.baseConfig.plugins = config.baseConfig.plugins.concat('react');
+			config.baseConfig.rules['react/jsx-indent-props'] = ['error', spaces];
+			config.baseConfig.rules['react/jsx-indent'] = ['error', spaces];
 		}
 	}
 
 	if (options.semicolon === false && !options.prettier) {
 		if (options.ts) {
-			config.rules['@typescript-eslint/semi'] = ['error', 'never'];
+			config.baseConfig.rules['@typescript-eslint/semi'] = ['error', 'never'];
 		} else {
-			config.rules.semi = ['error', 'never'];
+			config.baseConfig.rules.semi = ['error', 'never'];
 		}
 
-		config.rules['semi-spacing'] = ['error', {
+		config.baseConfig.rules['semi-spacing'] = ['error', {
 			before: false,
 			after: true
 		}];
 	}
 
 	if (options.ts) {
-		config.rules['unicorn/import-style'] = 'off';
-		config.rules['node/file-extension-in-import'] = 'off';
+		config.baseConfig.rules['unicorn/import-style'] = 'off';
+		config.baseConfig.rules['node/file-extension-in-import'] = 'off';
 
 		// Disabled because of https://github.com/benmosher/eslint-plugin-import/issues/1590
-		config.rules['import/export'] = 'off';
+		config.baseConfig.rules['import/export'] = 'off';
 
 		// Does not work when the TS definition exports a default const.
-		config.rules['import/default'] = 'off';
-	}
-
-	if (options.rules) {
-		Object.assign(config.rules, options.rules);
-	}
-
-	if (options.parser) {
-		config.baseConfig.parser = options.parser;
-	}
-
-	if (options.processor) {
-		config.baseConfig.processor = options.processor;
+		config.baseConfig.rules['import/default'] = 'off';
 	}
 
-	config.baseConfig.settings = options.settings || {};
 	config.baseConfig.settings['import/resolver'] = gatherImportResolvers(options);
 
 	return config;
 };
 
 const buildExtendsConfig = options => config => {
 	if (options.extends && options.extends.length > 0) {
-		// TODO: This logic needs to be improved, preferably use the same code as ESLint
-		// user's configs must be resolved to their absolute paths
 		const configs = options.extends.map(name => {
 			// Don't do anything if it's a filepath
 			if (pathExists.sync(name)) {
 				return name;
 			}
 
-			// Don't do anything if it's a config from a plugin
-			if (name.startsWith('plugin:')) {
-				return name;
-			}
-
-			if (name.startsWith('@')) {
+			// Don't do anything if it's a config from a plugin or an internal eslint config
+			if (name.startsWith('eslint:') || name.startsWith('plugin:')) {
 				return name;
 			}
 
-			if (!name.includes('eslint-config-')) {
-				name = `eslint-config-${name}`;
-			}
-
-			const returnValue = resolveFrom(options.cwd, name);
+			const returnValue = resolveFrom(options.cwd, normalizePackageName(name, 'eslint-config'));
```

Comment:
Is there any ESLint API we could use instead of handling this manually? For example, `resolveFrom` will not work when we eventually switch to ESM.

Reply:
ESLint uses `require.resolve` behind the scenes:

https://github.com/eslint/eslintrc/blob/62ea4bd74b78fbeff12ffb21f1f978817601d4d1/lib/shared/relative-module-resolver.js#L28

Reply:
Can you use https://nodejs.org/api/module.html#module_module_createrequire_filename ?

Reply:
https://github.com/eslint/eslintrc/pull/34